tests/scripted/check-namespace

```diff
@@ -0,0 +1,199 @@
+                        # blank out the range between opening and closing symbols
+                        for k in range(i, j + 1):
+                            results[k] = ''
+                            removed = removed + 1
+                        break
+                    elif not (results[j] == '' or results[j] == '#el' or (
+                                also_incs_and_namespace and results[j].startswith(
+                                    '#include'))):  # can only be '{', #if... etc. so we can't eliminate starting at 'i'
+                        break
+        # print(f'{last_size} -> {cur_size}')
+        # print(f'{cur_file}: {results}')
+
+
+for cur_file in files:
+    if cur_file.is_file():
+        results = namespaces.findall(cur_file.read_text(errors='replace'))
+        results_as_str = '\n'.join(results)
+        # print(f'{cur_file}: {results_as_str}')
+
+        if str(cur_file) not in known_no_namespace_lmms:
+            has_namespace_lmms.search(results_as_str) or error(cur_file, f'File has no namespace lmms')
+
+        debugfile = ''  # 'include/RemotePluginBase.h' # debugging
+
+        # remove comments
+        cur_text = cur_file.read_text(errors='replace')
+        cur_text = re.sub(r'(^|[^/])/\*(.|\n)*?\*/', r'\1', cur_text)
+        cur_text = re.sub(r'// namespace', r'//%namespace', cur_text, 0, re.MULTILINE)
+        cur_text = re.sub(r'#endif // ?([A-Z_]+)$', r'#endif //%\1', cur_text, 0, re.MULTILINE)
+        if str(cur_file) == debugfile:
+            print(cur_text)
+        cur_text = re.sub(r'//[^%].*$', r'', cur_text, 0, re.MULTILINE)
+        cur_text = re.sub(r'#endif //%([A-Z_]+)$', r'#endif // \1', cur_text, 0, re.MULTILINE)
+        cur_text = re.sub(r'//%namespace', r'// namespace', cur_text, 0, re.MULTILINE)
+
+        # get list of all #include, #if../#el../#endif and {, }, namespace ... {
+        results = all_braces_and_macros.findall(cur_text)
+
+        if str(cur_file) == debugfile:
+            print(cur_text)
+            print(f'{cur_file}: {results}')
+
+        # remove all ' ' and '\t' from results
+        results = [mystr.strip() for mystr in results]
+        results = [re.sub(r'#\s+', '#', mystr) for mystr in results]
+        results = [re.sub(r'\s+{', ' {', mystr) for mystr in results]
+
+        # eliminate non namespace parentheses and #if../#el../#endif
+        eliminate_parentheses(results, False)
+
+        # the remainder should be solely includes + >=0 namespaces
+        results_as_str = '\n'.join(results)
+        no_include_after_namespace.match(results_as_str) or error(cur_file,
+                                                                  f'File \'{cur_file}\' has includes inside/after a namespace: {results_as_str}')
+
+        if str(cur_file) == debugfile:
+            print(f'{cur_file}: {results}')
+
+        # now also remove #include and the namespaces - the resulting string must be empty
```

Comment:
Doesn't the C++ compiler catch that already?

Reply:
It does, but if the string is not empty here, the script has an error. E.g. it could have removed `namespace xyz {`, but not the matching `} // namspace xyz`. I had these issues during development of the script, so I thought I leave this assertion here to make sure the script works.